WordPressKit/FeatureFlagRemote.swift

```diff
@@ -9,17 +9,22 @@ open class FeatureFlagRemote: ServiceRemoteWordPressComREST {
     }
 
     open func getRemoteFeatureFlags(forDeviceId deviceId: String, callback: @escaping FeatureFlagResponseCallback) {
+        self.getRemoteFeatureFlags(params: .init(deviceId: deviceId), callback: callback)
+    }
 
+    open func getRemoteFeatureFlags(params: RemoteFeatureFlagsEndpointParams, callback: @escaping FeatureFlagResponseCallback) {
         let endpoint = "mobile/feature-flags"
         let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
+        var parameters: [String: AnyObject]?
 
-        let parameters: [String: AnyObject] = [
-            "device_id": deviceId as NSString,
-            "platform": "ios" as NSString,
-            "build_number": NSString(string: Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? "Unknown"),
-            "marketing_version": NSString(string: Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "Unknown"),
-            "identifier": NSString(string: Bundle.main.bundleIdentifier ?? "Unknown")
-        ]
+        do {
+            let encoder = JSONEncoder()
+            let data = try encoder.encode(params)
+            parameters = try JSONSerialization.jsonObject(with: data) as? [String: AnyObject]
+        } catch let error {
```

Comment:
[Suggestion] How about moving this to a helper function on `RemoteFeatureFlagsEndpointParams`? Something like this:
```
    func dictionaryRepresentation() throws -> [String: AnyObject]?  {
        let encoder = JSONEncoder()
        let data = try encoder.encode(self)
        return try JSONSerialization.jsonObject(with: data) as? [String: AnyObject]
    }
```

This way it can be used here and also in `DashboardServiceRemote.swift:L41`

Reply:
Pushed commits resolving this ✅